From 3ba63bacfcb5983d756e7346c7d66eaf36a18e49 Mon Sep 17 00:00:00 2001
From: Lin Ruizhe <linruizhe@huawei.com>
Date: Fri, 31 Dec 2021 15:59:21 +0800
Subject: [PATCH] bootparam: Add kabi_reserve in bootparam
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 3ba63bacfcb5983d756e7346c7d66eaf36a18e49


hulk inclusion
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MZU1
CVE: NA

---------------------------

Add kabi_resetve in struct acpi_rsdp_addr and setup_header

Signed-off-by: Lin Ruizhe <linruizhe@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/x86/include/uapi/asm/bootparam.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/include/uapi/asm/bootparam.h b/arch/x86/include/uapi/asm/bootparam.h
index 600a141c8805..b1dce768d073 100644
--- a/arch/x86/include/uapi/asm/bootparam.h
+++ b/arch/x86/include/uapi/asm/bootparam.h
@@ -42,6 +42,7 @@
 #include <linux/types.h>
 #include <linux/screen_info.h>
 #include <linux/apm_bios.h>
+#include <linux/kabi.h>
 #include <linux/edd.h>
 #include <asm/ist.h>
 #include <video/edid.h>
@@ -102,6 +103,7 @@ struct setup_header {
 	__u32	init_size;
 	__u32	handover_offset;
 	__u32	kernel_info_offset;
+	KABI_RESERVE(1)
 } __attribute__((packed));
 
 struct sys_desc_table {
-- 
2.26.2

